source/common/http/conn_manager_impl.cc

```diff
@@ -553,6 +553,9 @@ ConnectionManagerImpl::ActiveStream::ActiveStream(ConnectionManagerImpl& connect
                connection_manager.config_.scopedRouteConfigProvider() == nullptr)),
          "Either routeConfigProvider or scopedRouteConfigProvider should be set in "
          "ConnectionManagerImpl.");
+
+  stream_info_.setRequestIDUtils(connection_manager.config_.requestIDUtils());
```

Comment:
Why does stream info need the request ID utiltiies? I'm wondering if whatever stat it is being used for can be stored instead?

Reply:
Looks like it's used in two cases:
1. `AccessLog::TraceableRequestFilter` and `ConnectionManagerImpl::ActiveStream` to determine if request is traced (using `Tracing::HttpTracerUtility`)
2. `AccessLog::RuntimeFilter` uses it for making deterministic logging sampling decision

Given that `AccessLog` instances are shared across all L4/L5 connection managers, and `RequestIDUtils` is configured on per-manager basis we need a side channel for passing this information.

I'm worried about preserving derivatives (boolean `isTraced` and raw sampling number) as they can become out of sync if request id is changed mid-flight. It doesn't look like any existing extensions currently do it though.

Reply:
OK fair enough I think that's fine.